ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,163 @@
+    return getattr(anthropic, "__version__", "")
+
+
+config._add(
+    "anthropic",
+    {
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_ANTHROPIC_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "span_char_limit": int(os.getenv("DD_ANTHROPIC_SPAN_CHAR_LIMIT", 128)),
+    },
+)
+
+
+def _extract_api_key(instance: Any) -> str:
+    """
+    Extract and format LLM-provider API key from instance.
+    """
+    client = getattr(instance, "_client", "")
+    if client:
+        return getattr(client, "api_key", None)
+    return None
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    span = integration.trace(
+        pin,
+        "%s.%s" % (instance.__module__, instance.__class__.__name__),
+        submit_to_llmobs=True,
+        interface_type="chat_model",
+        provider="anthropic",
+        model=kwargs.get("model", ""),
+        api_key=_extract_api_key(instance),
+    )
+    chat_completions = None
+    try:
+        for message_idx, message in enumerate(chat_messages):
+            if isinstance(message, dict):
+                if isinstance(message.get("content", None), str):
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.0.content" % (message_idx),
+                        integration.trunc(message.get("content", "")),
+                    )
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.0.type" % (message_idx),
+                        "text",
+                    )
+                elif isinstance(message.get("content", None), list):
+                    for prompt_idx, prompt in enumerate(message.get("content", [])):
+                        if integration.is_pc_sampled_span(span):
+                            if prompt.get("type", None) == "text":
+                                span.set_tag_str(
+                                    "anthropic.request.messages.%d.%d.content" % (message_idx, prompt_idx),
+                                    integration.trunc(str(prompt.get("text", ""))),
+                                )
+                            elif prompt.get("type", None) == "image":
+                                span.set_tag_str(
+                                    "anthropic.request.messages.%d.%d.content" % (message_idx, prompt_idx),
+                                    "([IMAGE DETECTED])",
+                                )
+
+                        span.set_tag_str(
+                            "anthropic.request.messages.%d.%d.type" % (message_idx, prompt_idx),
+                            prompt.get("type", "text"),
+                        )
+                span.set_tag_str(
+                    "anthropic.request.messages.%d.role" % (message_idx),
+                    message.get("role", ""),
+                )
```

Comment:
When you generate a span with two tags `request.messages.0.0.content` and `request.messages.0.role`, how does it show up on Datadog's span tags UI?

Usually specifying tags to have numeric values inside will make the UI render it as an array, but I'm concerned adding a `role` field will mess up rendering. Would you be able to attach a screenshot? Thanks!

Reply:
oh wait, you are correct, the tags for role do not work. I think i will change the role and other tagging to be set on each content block